Userland/Libraries/LibTextCodec/Decoder.cpp

```diff
@@ -422,4 +434,19 @@ String TurkishDecoder::to_utf8(const StringView& input)
     return builder.to_string();
 }
 
+String ShiftJISEncoder::to_utf8(StringView const& input)
+{
+    StringBuilder builder;
+    for (size_t i = 0; i < input.length();) {
+        if (((u8)input[i]) < 0x80) {
+            builder.append_code_point(convert_shift_jis_to_utf8(input[i]));
+            i++;
+        } else {
+            builder.append_code_point(convert_shift_jis_to_utf8((input[i] << 8) | input[i + 1]));
+            i += 2;
+        }
+    }
```

Comment:
This is fine as well, but this might look a bit nicer:
```suggestion
    for (size_t i = 0; i < input.length(); ++i) {
        if (((u8)input[i]) < 0x80) {
            builder.append_code_point(convert_shift_jis_to_utf8(input[i]));
        } else {
            builder.append_code_point(convert_shift_jis_to_utf8((input[i] << 8) | input[i + 1]));
            ++i;
        }
    }
```

Reply:
Applied diff in current patchset.